### ISSUE TITLE
Why compute IPO loss using `average_log_prob=Ture`？

### ISSUE THREAD

**Issue by AIR-hl** [huggingface/trl#1677]
Why compute IPO loss using `average_log_prob=Ture`
In function `concatenated_forward`，when the `loss_type` equals 'ipo' the parameter `average_log_prob` will be set True, but according to the loss formula of IPO, the length `m` as a factor will be eliminated

![image](https://github.com/huggingface/trl/assets/66808901/973c1177-2759-4a84-bafe-fcff30c47f97)

``` 
all_logps = self.get_batch_logps(
    all_logits,
    concatenated_batch["concatenated_labels"],
    average_log_prob=self.loss_type == "ipo",
    is_encoder_decoder=self.is_encoder_decoder,
    label_pad_token_id=self.label_pad_token_id,
)
```

**Comment by QiyaoWei** [comment]
This has been discussed in [multiple](https://github.com/huggingface/trl/pull/1265) [github](https://github.com/h2oai/h2o-llmstudio/issues/580) [issues](https://github.com/huggingface/trl/issues/1294), and I believe the answer stems from the discussion that Huggingface had with the IPO authors [here](https://huggingface.co/blog/pref-tuning) "After consulting with the authors of the IPO paper, we discovered that the implementation of IPO in TRL was incorrect; in particular, the loss over the log-likelihoods of the completions needs to be averaged instead of summed. We have added a fix..."

**Comment by AIR-hl** [comment]
> This has been discussed in [multiple](https://github.com/huggingface/trl/pull/1265) [github](https://github.com/h2oai/h2o-llmstudio/issues/580) [issues](https://github.com/huggingface/trl/issues/1294), and I believe the answer stems from the discussion that Huggingface had with the IPO authors [here](https://huggingface.co/blog/pref-tuning) "After consulting with the authors of the IPO paper, we discovered that the implementation of IPO in TRL was incorrect; in particular, the loss over the log-likelihoods of the completions needs to be averaged instead of summed. We have added a fix..."

Please forgive me, I don't find any formula using average log-likelihood in paper

**Comment by AIR-hl** [comment]
> This has been discussed in [multiple](https://github.com/huggingface/trl/pull/1265) [github](https://github.com/h2oai/h2o-llmstudio/issues/580) [issues](https://github.com/huggingface/trl/issues/1294), and I believe the answer stems from the discussion that Huggingface had with the IPO authors [here](https://huggingface.co/blog/pref-tuning) "After consulting with the authors of the IPO paper, we discovered that the implementation of IPO in TRL was incorrect; in particular, the loss over the log-likelihoods of the completions needs to be averaged instead of summed. We have added a fix..."

@QiyaoWei Please forgive me, I don't find any formula using average log-likelihood in paper
